pkg/config/generate.go

```diff
@@ -388,3 +388,18 @@ func Generate(cs *api.OpenShiftManagedCluster, pluginConfig api.PluginConfig) (e
 
 	return
 }
+
+func regenerate(cert certificate) bool {
+	crt := *cert.cert
+	if crt == nil {
+		return false
+	}
+
+	if !reflect.DeepEqual(cert.dnsNames, crt.DNSNames) ||
+		!reflect.DeepEqual(cert.extKeyUsage, crt.ExtKeyUsage) ||
+		!reflect.DeepEqual(cert.cn, crt.Subject.CommonName) ||
+		!reflect.DeepEqual(cert.ipAddresses, crt.IPAddresses) {
+		return false
```

Comment:
We might need to do some logic to check signing key too. but this is more complex.

Reply:
also consider the organization

Reply:
the signing key shouldn't be too bad - if !selfSign, then call CheckSignatureFrom (the code you deleted above)

Reply:
ok, signing key was not hard. IP address was. Can you advise anything on this, please?